lib/cxx-api/tests/config.cc

```diff
@@ -73,4 +73,63 @@ TEST_CASE_METHOD(Fixture, "config solve", "[cxx][config][solve]") {
     REQUIRE(models == res);
 }
 
+TEST_CASE_METHOD(Fixture, "config extend", "[cxx][config][extend]") {
+    class ConfigValue {
+      public:
+        [[nodiscard]] auto get() const -> std::optional<std::string> { return value_; }
+        void set(std::string_view value) { value_ = value; }
+
+      private:
+        std::optional<std::string> value_;
+    };
+    class ConfigArray {
+      public:
+        ConfigArray(std::vector<std::optional<std::string>> &values) : values_{&values} {}
+        [[nodiscard]] auto size() const -> size_t { return values_->size(); }
+
+      private:
+        std::vector<std::optional<std::string>> *values_;
+    };
+    class ConfigArrayValue {
+      public:
+        ConfigArrayValue(std::vector<std::optional<std::string>> &values) : values_{&values} {}
+        [[nodiscard]] auto get(std::optional<size_t> index) const -> std::optional<std::string> {
+            return values_->at(index ? *index : 0);
```

Comment:
Nitpick: Consider using `value_or` here and below.

Reply:
Changed.